codex-rs/core/tests/suite/mod.rs

```diff
@@ -1,4 +1,30 @@
 // Aggregates all former standalone integration tests as modules.
+use codex_arg0::arg0_dispatch_or_else;
+use codex_arg0::prepend_path_entry_for_codex_aliases;
+use ctor::ctor;
+use tempfile::TempDir;
+
+// This code runs before any other tests are run.
+// It allows the test binary to behave like codex and dispatch to apply_patch and codex-linux-sandbox
+// based on the arg0.
+// NOTE: this doesn't work on ARM
+#[ctor]
+pub static CODEX_ALIASES_TEMP_DIR: TempDir = unsafe {
+    let mut should_exit = true;
+
+    #[allow(clippy::unwrap_used)]
+    arg0_dispatch_or_else(|_| async {
```

Comment:
`async` starts the tokio runtime, which I do not think we should be doing pre-main